ws281xspi.pro

```diff
@@ -0,0 +1,17 @@
+include($$[QT_INSTALL_PREFIX]/include/nymea/plugin.pri)
+
+QMAKE_PRE_LINK = scons -C $$_PRO_FILE_PWD_/rpi_ws281x libws2811.a
```

Comment:
Doesn't look like this actually does what you want, does it?

afaict the plugin is now built first with make, and then you're running scons on it again just to print that make already did everything...  Maybe I'm wrong... never used scons so far.

Reply:
No, this file actually builds the library so it can be linked with the plugin in the next step. If the conclusion is to drop scons, this line would of course disappear also.

Reply:
```
14:03:04: Running steps for project ws281xspi...
14:03:04: Starting: "/usr/lib/qt5/bin/qmake" /home/micha/Develop/nymea-plugin-ws281x/ws281xspi.pro -spec linux-g++ CONFIG+=debug CONFIG+=qml_debug PLUGIN_PRI=/home/micha/Develop/nymea/libnymea/integrations/plugin.pri
Project MESSAGE: ****
14:03:04: The process "/usr/lib/qt5/bin/qmake" exited normally.
14:03:04: Starting: "/usr/bin/make" -f /home/micha/Develop/build-ws281xspi-Desktop-Debug/Makefile qmake_all
make: Nothing to be done for 'qmake_all'.
14:03:04: The process "/usr/bin/make" exited normally.
14:03:04: Starting: "/usr/bin/make" -j8
g++ -c -pipe -Werror -std=c++11 -g -g -Wall -W -D_REENTRANT -fPIC -DQT_QML_DEBUG -DQT_PLUGIN -DQT_GUI_LIB -DQT_CORE_LIB -I../nymea-plugin-ws281x -I. -I../nymea-plugin-ws281x/rpi_ws281x -isystem /usr/include/nymea/ -isystem /usr/include/x86_64-linux-gnu/qt5 -isystem /usr/include/x86_64-linux-gnu/qt5/QtGui -isystem /usr/include/x86_64-linux-gnu/qt5/QtCore -I. -I/usr/lib/x86_64-linux-gnu/qt5/mkspecs/linux-g++ -o integrationpluginws281xspi.o ../nymea-plugin-ws281x/integrationpluginws281xspi.cpp
nymea-plugininfocompiler ../nymea-plugin-ws281x/integrationpluginws281xspi.json --extern extern-plugininfo.h
g++ -pipe -Werror -std=c++11 -g -g -Wall -W -dM -E -o moc_predefs.h /usr/lib/x86_64-linux-gnu/qt5/mkspecs/features/data/dummy.cpp
/usr/lib/qt5/bin/moc -DQT_QML_DEBUG -DQT_PLUGIN -DQT_GUI_LIB -DQT_CORE_LIB --include /home/micha/Develop/build-ws281xspi-Desktop-Debug/moc_predefs.h -I/usr/lib/x86_64-linux-gnu/qt5/mkspecs/linux-g++ -I/home/micha/Develop/nymea-plugin-ws281x -I/home/micha/Develop/nymea-plugin-ws281x/rpi_ws281x -I/usr/include/nymea -I/usr/include/x86_64-linux-gnu/qt5 -I/usr/include/x86_64-linux-gnu/qt5/QtGui -I/usr/include/x86_64-linux-gnu/qt5/QtCore -I. -I/home/micha/Develop/nymea/libnymea -I/usr/include/c++/9 -I/usr/include/x86_64-linux-gnu/c++/9 -I/usr/include/c++/9/backward -I/usr/lib/gcc/x86_64-linux-gnu/9/include -I/usr/local/include -I/usr/include/x86_64-linux-gnu -I/usr/include ../nymea-plugin-ws281x/integrationpluginws281xspi.h -o moc_integrationpluginws281xspi.cpp
/usr/lib/qt5/bin/moc -DQT_QML_DEBUG -DQT_PLUGIN -DQT_GUI_LIB -DQT_CORE_LIB --include /home/micha/Develop/build-ws281xspi-Desktop-Debug/moc_predefs.h -I/usr/lib/x86_64-linux-gnu/qt5/mkspecs/linux-g++ -I/home/micha/Develop/nymea-plugin-ws281x -I/home/micha/Develop/nymea-plugin-ws281x/rpi_ws281x -I/usr/include/nymea -I/usr/include/x86_64-linux-gnu/qt5 -I/usr/include/x86_64-linux-gnu/qt5/QtGui -I/usr/include/x86_64-linux-gnu/qt5/QtCore -I. -I/home/micha/Develop/nymea/libnymea -I/usr/include/c++/9 -I/usr/include/x86_64-linux-gnu/c++/9 -I/usr/include/c++/9/backward -I/usr/lib/gcc/x86_64-linux-gnu/9/include -I/usr/local/include -I/usr/include/x86_64-linux-gnu -I/usr/include ../nymea-plugin-ws281x/ws281xspicontroller.h -o moc_ws281xspicontroller.cpp
g++ -c -pipe -Werror -std=c++11 -g -g -Wall -W -D_REENTRANT -fPIC -DQT_QML_DEBUG -DQT_PLUGIN -DQT_GUI_LIB -DQT_CORE_LIB -I../nymea-plugin-ws281x -I. -I../nymea-plugin-ws281x/rpi_ws281x -isystem /usr/include/nymea/ -isystem /usr/include/x86_64-linux-gnu/qt5 -isystem /usr/include/x86_64-linux-gnu/qt5/QtGui -isystem /usr/include/x86_64-linux-gnu/qt5/QtCore -I. -I/usr/lib/x86_64-linux-gnu/qt5/mkspecs/linux-g++ -o ws281xspicontroller.o ../nymea-plugin-ws281x/ws281xspicontroller.cpp
g++ -c -pipe -Werror -std=c++11 -g -g -Wall -W -D_REENTRANT -fPIC -DQT_QML_DEBUG -DQT_PLUGIN -DQT_GUI_LIB -DQT_CORE_LIB -I../nymea-plugin-ws281x -I. -I../nymea-plugin-ws281x/rpi_ws281x -isystem /usr/include/nymea/ -isystem /usr/include/x86_64-linux-gnu/qt5 -isystem /usr/include/x86_64-linux-gnu/qt5/QtGui -isystem /usr/include/x86_64-linux-gnu/qt5/QtCore -I. -I/usr/lib/x86_64-linux-gnu/qt5/mkspecs/linux-g++ -o moc_ws281xspicontroller.o moc_ws281xspicontroller.cpp
g++ -c -pipe -Werror -std=c++11 -g -g -Wall -W -D_REENTRANT -fPIC -DQT_QML_DEBUG -DQT_PLUGIN -DQT_GUI_LIB -DQT_CORE_LIB -I../nymea-plugin-ws281x -I. -I../nymea-plugin-ws281x/rpi_ws281x -isystem /usr/include/nymea/ -isystem /usr/include/x86_64-linux-gnu/qt5 -isystem /usr/include/x86_64-linux-gnu/qt5/QtGui -isystem /usr/include/x86_64-linux-gnu/qt5/QtCore -I. -I/usr/lib/x86_64-linux-gnu/qt5/mkspecs/linux-g++ -o moc_integrationpluginws281xspi.o moc_integrationpluginws281xspi.cpp
scons -C /home/micha/Develop/nymea-plugin-ws281x/rpi_ws281x libws2811.a
scons: Entering directory `/home/micha/Develop/nymea-plugin-ws281x/rpi_ws281x'
scons: Reading SConscript files ...
scons: done reading SConscript files.
scons: Building targets ...
scons: `libws2811.a' is up to date.
scons: done building targets.
rm -f libnymea_integrationpluginws281xspi.so
g++ -std=c++11 -shared -o libnymea_integrationpluginws281xspi.so integrationpluginws281xspi.o ws281xspicontroller.o moc_integrationpluginws281xspi.o moc_ws281xspicontroller.o  -L/home/micha/Develop/nymea-plugin-ws281x/rpi_ws281x/ -lws2811 -lnymea /usr/lib/x86_64-linux-gnu/libQt5Gui.so /usr/lib/x86_64-linux-gnu/libQt5Core.so /usr/lib/x86_64-linux-gnu/libGL.so -lpthread    
14:03:06: The process "/usr/bin/make" exited normally.
14:03:06: Elapsed time: 00:02.
```

To me this really looks like it's running make and when make is done it runs scons which just says "scons: `libws2811.a' is up to date."

Reply:
Oh... stupid me.. this just builds the submodule stuff... Right...